ForgeUnity/Assets/BeardedManStudios/Scripts/NetworkManager.cs

```diff
@@ -711,5 +728,76 @@ public GameObject GetGameObjectByNetworkId(uint id)
 
             return ((NetworkBehavior)foundNetworkObject.AttachedBehavior).gameObject;
         }
-    }
+
+		public virtual void SceneUnloaded(Scene scene)
+		{
+			// The NetworkManager has not yet been initialized with a Networker.
+			if (!initialized)
+				return;
+
+			loadedScenes.Remove(scene.buildIndex);
+
+			// Send buildindex and 2 refering to
+			BMSByte data = ObjectMapper.BMSByte(scene.buildIndex, UnloadSceneCommand);
+
+			Binary frame = new Binary(Networker.Time.Timestep, false, data, Networker is IServer ? Receivers.All : Receivers.Server, MessageGroupIds.VIEW_CHANGE, false);
+
+			// Send the binary frame to either the server or the clients
+			SendFrame(Networker, frame);
+		}
+
+		public void UnloadSceneAdditive(int buildIndex)
+		{
+			if (!initialized || buildIndex < 0)
+				return;
+
+			Scene scene = SceneManager.GetSceneByBuildIndex(buildIndex);
+
+			if (scene.IsValid() && scene.isLoaded)
+			{
+				UnloadSceneNetworkObjects(buildIndex);
+
+				SceneManager.UnloadSceneAsync(buildIndex);
+			}
+		}
+
+		public void UnloadSceneAdditive(string sceneName)
```

Comment:
This method and the one above seem to behave differently. This one does not check if the scene is loaded or not while the other one does.

Maybe you could create one method that both of these could call to unload the scene?
Or maybe one of them could call the other one.

Thinking about the one with the sceneName argument to maybe get the build index of the scene and call the other method to unload?

Or maybe have a third method that both would call that take a scene object as it's argument? :thinking:

Reply:
Look at my new commit, just forgot to remove isloaded